analyzer-java/src/main/java/com/infosupport/ldoc/analyzerj/AnalysisVisitor.java

```diff
@@ -94,66 +99,140 @@ public List<Description> visit(ClassOrInterfaceDeclaration n, Analyzer arg) {
     baseTypes.addAll(resolve(n.getExtendedTypes()));
     baseTypes.addAll(resolve(n.getImplementedTypes()));
 
-    return List.of(new TypeDescription(
-        n.isInterface() ? TypeType.INTERFACE : TypeType.CLASS,
-        combine(n.getModifiers()),
-        n.getFullyQualifiedName().orElseThrow(),
-        baseTypes,
-        n.getComment().flatMap(c -> c.accept(this, arg).stream().findFirst()).orElse(null),
-        select(n.getMembers(), BodyDeclaration::isConstructorDeclaration, arg),
-        select(n.getMembers(), BodyDeclaration::isMethodDeclaration, arg),
-        visit(n.getAnnotations(), arg)));
+    return List.of(
+        new TypeDescription(
```

Comment:
Maybe, it would be better code maintainability wise. On the other hand, we have implemented all types in Java so not sure whether it would be a real benefit. So I would say lets make this dependent on possible future advice of SonarQube